org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/xsd/participants/XSDErrorCode.java

```diff
@@ -156,7 +157,10 @@ public static Range toLSPRange(XMLLocator location, XSDErrorCode code, Object[]
 			String attrValue = getString(arguments[0]);
 			return XMLPositionUtility.selectAttributeValueByGivenValueAt(attrValue, offset, document);
 		}
-		case EmptyTargetNamespace:
+		case schema_reference_4: {
+			return XMLPositionUtility.selectAttributeValueAt("schemaLocation", offset, document);

```

Comment:
replace `"schemaLocation" `with `XSDUtils.SCHEMA_LOCATION_ATTR`

Reply:
This change reverts the fix to the problem, i.e. the highlighting occurs at the closing bracket

Reply:
Sorry I don't understand, you mean if you write

```java
case schema_reference_4: {
	return XMLPositionUtility.selectAttributeValueAt(XSDUtils.SCHEMA_LOCATION_ATTR, offset, document);
```

it doesn't work?

Reply:
On my side, the change doesn't work (i.e. I see the old problem again)